### DISCUSSION TITLE
How to get the whole polynomial basis P(n) of a first degree tetrahedral element?

### DISCUSSION THREAD

**Discussion by prusso630** [mscroggs/symfem#242] [Q&A] [answered]
```
import symfem
from symfem.symbols import x

vs = [(0,0,1), (0, -1, 0), (-1, 0,0), (0, 1,0)]
element = symfem.create_element("tetrahedron", "Lagrange", 1)
ref = symfem.create_reference("tetrahedron", vertices=vs) 
print(element.get_basis_functions()) 
print('done')
```

I see that there is an element.get_basis_functions():
```
[-x - y - z + 1, x, y, z]
```

Somehow I have the impression that there is a whole polynomial basis for a tetrahedron that should correspond to the number of elements of the tetrahedron and it's corresponding degree. It seems to me that element.get_basis_functions() is just at the root of the element not the whole thing. Is that correct or not? 

From Josep Birnic's I have a 2D example for the sigma sum to look to:
P_n(x,y) = SigmaSum_k=0_n  alpha_k*x^iy^i, +j<=k,

The number of terms is said to be (n+1)(n+2)/2 for a 2D for a 3D its not stated in my book?

I don't have a good example yet of exactly what things are in 3D. P_n being the number of elements (x,y,z) in the whole set.

Is there a way to get the whole P_n from the co-ordinate list of the tet at said vertices?

**Comment by mscroggs** [comment, marked as answer]
For a Lagrange element of degree n on a tetrahedron, there are (n+1)(n+2)(n+3)/6 polynomials in the element (you can find this and lots more information on the [DefElement Lagrange page](https://defelement.com/elements/lagrange.html)).

`element.get_basis_functions()` gets the finite element basis functions of the element. You can also use `element.get_polynomial_basis()` to get a basis of the polynomial space on the cell. The results of both these functions will span the same set of polynomials.
